spring-integration-zeromq/src/test/java/org/springframework/integration/zeromq/dsl/ZeroMqDslTests.java

```diff
@@ -0,0 +1,153 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.zeromq.dsl;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.time.Duration;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.Test;
+import org.zeromq.SocketType;
+import org.zeromq.ZContext;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.Transformers;
+import org.springframework.integration.dsl.context.IntegrationFlowContext;
+import org.springframework.integration.zeromq.ZeroMqHeaders;
+import org.springframework.integration.zeromq.ZeroMqProxy;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+import org.springframework.util.SocketUtils;
+
+/**
+ * @author Artem Bilan
+ *
+ * @since 5.4
+ */
+@SpringJUnitConfig
+@DirtiesContext
+public class ZeroMqDslTests {
+
+	private static final int PROXY_PUB_PORT = SocketUtils.findAvailableTcpPort();
+
+	@Autowired
+	ZContext context;
+
+	@Autowired
+	@Qualifier("publishToZeroMqPubSubFlow.input")
+	MessageChannel publishToZeroMqPubSubFlowInput;
+
+	@Autowired
+	ZeroMqProxy subPubZeroMqProxy;
+
+	@Autowired
+	ZeroMqProxy pullPushZeroMqProxy;
+
+	@Autowired
+	IntegrationFlowContext integrationFlowContext;
+
+	@Test
+	void testZeroMqDslIntegration() throws InterruptedException {
+		BlockingQueue<Message<?>> results = new LinkedBlockingQueue<>();
+
+		await().until(() -> this.subPubZeroMqProxy.getBackendPort() > 0);
+
+		for (int i = 0; i < 2; i++) {
+			IntegrationFlow consumerFlow =
+					IntegrationFlows.from(
+							ZeroMq.inboundChannelAdapter(this.context, SocketType.SUB)
+									.connectUrl("tcp://localhost:" + this.subPubZeroMqProxy.getBackendPort())
+									.topics("someTopic")
+									.consumeDelay(Duration.ofMillis(100)))
+							.channel(ZeroMq.zeroMqChannel(this.context).zeroMqProxy(this.pullPushZeroMqProxy))
+							.transform(Transformers.objectToString())
+							.handle(results::offer)
+							.get();
+
+			this.integrationFlowContext.registration(consumerFlow).register();
+		}
+
+		// Give it some time to connect and subscribe
+		Thread.sleep(2000);
```

Comment:
Should we publish an `ApplicationEvent`?

Reply:
Well, we can't: the `ZMQ.Socket.subscribe()` is an async operation and there is no way on the `PUB` side to track it.
We can do that with `XPUB`, but that's not an intention of our solution...